### DISCUSSION TITLE
Conflicting dependecy requirement bettween spacy(v3.4.4) and spacy-transformer(v1.2.0)

### DISCUSSION THREAD

**Discussion by bkp4113** [explosion/spaCy#12330] [Help: Installation] [answered]
We use spacy v3.4.4 with transformer as an extra and have part of custom install using pip with spacy-transformer(v1.2.0) on our sage instance. However when using the pipenv, We can't resolve the dependency since the spacy(v3.4.4) limits the spacy-transformers dependency to <1.2.0. Only spacy(v3.5.0) allows/opens up spacy_transformers to >=1.1.2,<1.3.0. (Our models are trained with spacy v3.4 and spacy-transformer v1.2.0). I am not sure if this purposely or just missed out. Please advise.

Please also advise if we can we use spacy v3.4.4 and spacy-transformers v1.2.0 ?

Note below setup file conflicts
- spacy (3.4.4) [setup.cfg](https://github.com/explosion/spaCy/blob/77833bfef9d96fc49e98b9ccaf67e5c726aec75d/setup.cfg#L76)
- spacy-transformers (1.2.0) [setup.cfg](https://github.com/explosion/spacy-transformers/blob/8b587a4c68a26436fdc8ee1b1a4b48852bfbc8e6/setup.cfg#L34)

**Comment by adrianeboyd** [comment, marked as answer]
It's intentional that `spacy-transformers` v1.2+ is only compatible with `spacy` v3.5+. The underlying concern is that `spacy-transformers` v1.2.x models aren't backwards-compatible with `spacy-transformers` v1.1.x.

We didn't want `pip install spacy[transformers]>=3.4.0,<3.5.0` to potentially install a version of `spacy-transformers` where the saved models aren't compatible with a venv that was installed with the same requirement at an earlier date.

Since it might point to a problem on our side: how did you end up with spacy v3.4.4 and spacy-transformers v1.2.0 in the first place? `pip` won't 100% prevent you from installing conflicting packages, but it usually either upgrades/downgrades in the background or there's a red error like:

```none
$ pip install spacy==3.4.4 spacy-transformers==1.2.2
ERROR: Cannot install spacy-transformers==1.2.2 and spacy==3.4.4 because these package versions have conflicting dependencies.
```

Due to bug fixes in `spacy-transformers` v1.2.2, I would recommend either:

* use `spacy` v3.4.x and retrain your models with `spacy-transformers` v1.1.x
* use `spacy` v3.5.x with `spacy-transformers` v1.2.2+

(I wouldn't recommend it because of the dependency conflicts, but I don't think there should be any technical problems with using `spacy` v3.4.4 with `spacy-transformers` v1.2.0.)

**Comment by bkp4113** [comment]
Thank you for clarification @adamhadani.
